### ISSUE TITLE
Error in rat test Command and Google Earth Engine Connection Issue

### ISSUE THREAD

**Issue by jamesYu365** [UW-SASWE/RAT#119]
**Issue 1: TypeError during rat test command execution**
When running the rat test command:
```
rat test -d ./rat_project/ -b GUNNISON -s ./rat_project/params/secrets.ini
```
I encountered the following error message:
```
object address  : 0x7f5ab8cd83a0
object refcount : 3
object type     : 0x5570a836d1c0
object type name: TypeError
object repr     : TypeError("unsupported operand type(s) for -: 'datetime.date' and 'datetime.datetime'")
lost sys.stderr
```
Could you provide guidance on resolving this issue?

**Issue 2: Failed to Connect to Earth Engine**
I followed the instructions for setting up Google Earth Engine (GEE) credentials as outlined in the [Getting Ready](https://rat-satellitedams.readthedocs.io/en/latest/QuickStart/GettingReady/) and configured the secrets.ini file based on the [RAT Secrets File](https://rat-satellitedams.readthedocs.io/en/latest/Configuration/secrets/).

However, the log file indicates that the connection to Earth Engine failed with the message:

```
2024-11-20 09:05:21 INFO:init_logger>> -------------------- INITIALIZED RAT-logs LOG ------------------
2024-11-20 09:05:21 INFO:init_logger>> TIME: 2024-11-20 09:05:21.447852
2024-11-20 09:05:21 INFO:init_logger>> LOG LEVEL: DEBUG
2024-11-20 09:05:21 INFO:init_logger>> Logging To Console: False
2024-11-20 09:05:21 INFO:init_logger>> LOG FILE: /home/yubin/rat_project/data/test_output/runs/logs/RAT_run-20241120-090521.log
2024-11-20 09:05:21 INFO:init_logger>> NOTIFY: False
2024-11-20 09:05:21 INFO:init_logger>> ----------------------------------------------------------

2024-11-20 09:05:21 DEBUG:run_rat>> Initiating Dask Client ...
2024-11-20 09:05:23 DEBUG:run_rat>> Started client with 72 workers. Dashboard link: http://127.0.0.1:8787/status
2024-11-20 09:05:23 INFO:run_rat>> Checking earth engine credentials
2024-11-20 09:05:25 INFO:run_rat>> Failed to connect to Earth engine. Wrong credentials. If you want to use Surface Area Estimations from RAT, please update the EE credentials.
2024-11-20 09:05:25 INFO:run_rat>> ############## Starting RAT for gunnison #################
```
Below is the configuration in my secrets.ini file:

```
# Enter service account (account_email, path to private json key) for Google Earth Engine
[ee]
service_account=rat-china-yubin@rat-china-442207.iam.gserviceaccount.com
key_file=/home/yubin/rat_project/params/rat-china-442207-7949a2af3cce.json
```
Could you please assist in resolving this connection issue with Google Earth Engine?

**Comment by SanchitMinocha** [comment]
Thank you for raising the issue. It has been resolved in v3.0.15.

**Comment by SarathUW** [comment]
Note: This also fixes RAT-CWRDM issues with surface area computation
